src/bin/wc.rs

```diff
@@ -39,7 +41,63 @@ static MAN_PAGE: &'static str = r#"
         This program was written mainly by Alice Maz.
 "#;
 
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, Default)]
+struct Counter {
+    lines: u64,
+    words: u64,
+    bytes: u64,
+}
+
+impl Counter {
+    #[inline]
+    fn new<T: std::io::Read>(input: T) -> Self {
```

Comment:
I don't see why not? I just move over the `do_count` function to be the constructor.